src/sftp.rs

```diff
@@ -371,11 +393,36 @@ impl Sftp {
             Err(Error::from_errno(rc))
         }
     }
+
+    fn unwrap_raw_or_err(&self) -> Result<*mut raw::LIBSSH2_SFTP, Error> {
```

Comment:
FWIW, I found this name a bit confusing when seeing it used before reaching this function definition.
The `unwrap` implies the potential for a panic but this function isn't applying an `Option::unwrap` or a `Result::unwrap` style operation.
I'd suggest naming it something like `get_raw` instead; that way it's clear what the intent is, and the `Result` return type already indicates that it might yield an error.

Reply:
Ok, I was trying to refer to the unwrap_or/unwrap_or_else operations that don't panic. I changed it to get_raw as you suggested.